pkg/appgw/backendaddresspools.go

```diff
@@ -89,15 +90,28 @@ func (c *appGwConfigBuilder) newBackendPoolMap(cbCtx *ConfigBuilderContext) map[
 	_, _, serviceBackendPairMap, _ := c.getBackendsAndSettingsMap(cbCtx)
 	for backendID, serviceBackendPair := range serviceBackendPairMap {
 		backendPoolMap[backendID] = &defaultPool
-		if pool := c.getBackendAddressPool(backendID, serviceBackendPair, addressPools); pool != nil {
+		if backendID.isLDPBackend() {
+			ldpName := backendID.Backend.Resource.Name
+			ldp, err := c.k8sContext.GetLoadDistributionPolicy(backendID.Namespace, ldpName)
+			if err != nil { //skip if we cannot get LDP
+				continue
+			}
+			for _, target := range ldp.Spec.Targets {
+				serviceID := secretIdentifier{
+					Namespace: backendID.Namespace,
+					Name:      target.Backend.Service.Name,
+				}
+				c.getBackendAddressPool(backendID, serviceIdentifier(serviceID), serviceBackendPair, addressPools)
```

Comment:
ah yes, this enitre section should just skip over backends that have LDPs entirely. this is because it does not make sense to map a backendID to a backendAddressPool if it contains an LDP. 

Instead the way we deal with backend's that reference an LDP is by generating LDP ID. this logic can accordingly be deleted